### PR TITLE
docs: adds decision record about default datasource

### PR DESCRIPTION
**Pull request by wolf4ood**
## What this PR changes/adds

Adds decision about default datasource

## Why it does that

_Briefly state why the change was necessary._

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Ref #3337

### CODE REVIEW

**Review by ndr-brt — APPROVED** [review]

File: docs/developer/decision-records/2023-08-01-default-datasource/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Default Datasource configuration
+
+## Decision
+
+All SQL store extensions, if the specific datasource name is not configured should fall back to the default one.
+
+## Rationale
+
+Each SQL store implementation as of today needs to have at least one mandatory configuration
```
Comment (jimmarino) [inline docs/developer/decision-records/2023-08-01-default-datasource/README.md:9]:
```suggestion
Each SQL store implementation currently must have at least one mandatory configuration
```

File: docs/developer/decision-records/2023-08-01-default-datasource/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+parameter `edc.datasource.<entity>.name` and then in case additional parameters as `url`, `user`, `password` and so on.
+
+This is powerful for fine-grained configuration and deployment. For simpler case and scenario all the store should
```
Comment (jimmarino) [inline docs/developer/decision-records/2023-08-01-default-datasource/README.md:12]:
```suggestion
This is powerful for fine-grained configuration and deployment. For simpler cases and scenario all the stores should
```

File: docs/developer/decision-records/2023-08-01-default-datasource/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+default to a common datasource name, that will be easier to configure.
+
+## Approach
+
+All the SQL store extension will default to a common datasource name defined
```
Comment (jimmarino) [inline docs/developer/decision-records/2023-08-01-default-datasource/README.md:17]:
```suggestion
All the SQL store extensions will default to a common datasource name defined
```

**Review by jimmarino — APPROVED** [review]

**Comment by wolf4ood** [comment]
Will amend this in the PR

**Pull request merged**